applications/GeoMechanicsApplication/custom_workflows/dgeosettlement.cpp

```diff
@@ -59,6 +106,22 @@ int KratosGeoSettlement::RunStage(const std::filesystem::path&            rWorki
         KRATOS_INFO("KratosGeoSettlement") << "Read the materials from " << material_file_path << std::endl;
     }
 
+    std::vector<std::reference_wrapper<Process>> process_references;
+    std::vector<std::unique_ptr<Process>> process_unique_ptrs;
```

Comment:
@avdg81 The reference wrappers seem a bit cumbersome here. Maybe we can just pass an std::vector<std::unique_ptr<Process>>& to the TimeLoopExecutor?

Reply:
Changed it to the weak_ptr alternative we discussed. Let me know what you think!